src/libraries/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -193,35 +185,15 @@ internal static class UriHelper
             for (; i < stringToEscape.Length && (c = stringToEscape[i]) <= 0x7F && noEscape[c]; i++) ;
             if (i == stringToEscape.Length)
             {
-                if (dest != null)
-                {
-                    EnsureCapacity(dest, destPos, stringToEscape.Length);
-                    stringToEscape.CopyTo(dest.AsSpan(destPos));
-                    destPos += stringToEscape.Length;
-                }
-
-                return dest;
+                dest.Append(stringToEscape);
             }
+            else
+            {
+                dest.Append(stringToEscape.Slice(0, i));
 
-            // Otherwise, create a ValueStringBuilder to store the escaped data into,
-            // append to it all of the noEscape chars we already iterated through, and
-            // escape the rest into the ValueStringBuilder.
-            var vsb = new ValueStringBuilder(stackalloc char[256]);
-            vsb.Append(stringToEscape.Slice(0, i));
-            EscapeStringToBuilder(stringToEscape.Slice(i), ref vsb, noEscape, checkExistingEscaped);
-
-            // Finally update dest with the result.
-            EnsureCapacity(dest, destPos, vsb.Length);
-            vsb.TryCopyTo(dest.AsSpan(destPos), out int charsWritten);
-            destPos += charsWritten;
-            return dest;
+                ReadOnlySpan<bool> noEscapeCopy = MemoryMarshal.CreateReadOnlySpan(ref MemoryMarshal.GetReference(noEscape), UnreservedReservedTable.Length);
```

Comment:
This is a hack to avoid code analysis from preventing us passing the noEscape span as a parameter with

`CS8352	Cannot use local 'noEscape' in this context because it may expose referenced variables outside of their declaration scope`
and
`CS8350	This combination of arguments to 'UriHelper.EscapeStringToBuilder(ReadOnlySpan<char>, ref ValueStringBuilder, ReadOnlySpan<bool>, bool)' is disallowed because it may expose variables referenced by parameter 'noEscape' outside of their declaration scope`

Is this still legal from the runtime's perspective? Can it be avoided?
@stephentoub @jkotas

Reply:
It is ok from the runtime's perspective.

The only way to avoid this today that I think of is to stackalloc the flags in the same method that they are used in.